examples/apps/ping/src/icmpv4.adb

```diff
@@ -196,15 +201,29 @@ is
       RFLX.IPv4.Packet.Initialize_Payload (IP_Context);
       if RFLX.IPv4.Contains.ICMP_Message_In_Packet_Payload (IP_Context) then
          RFLX.IPv4.Contains.Switch_To_Payload (IP_Context, ICMP_Context);
-         RFLX.ICMP.Message.Set_Tag (ICMP_Context, RFLX.ICMP.Echo_Request);
-         RFLX.ICMP.Message.Set_Code_Zero (ICMP_Context, 0);
-         RFLX.ICMP.Message.Set_Checksum (ICMP_Context,
-                                         Checksum.Echo_Request_Reply_Checksum
-                                            (RFLX.ICMP.Echo_Request,
-                                             0, 0, Sequence, Data));
-         RFLX.ICMP.Message.Set_Identifier (ICMP_Context, 0);
-         RFLX.ICMP.Message.Set_Sequence_Number (ICMP_Context, Sequence);
-         RFLX.ICMP.Message.Set_Data (ICMP_Context, Data);
+         declare
+            use RFLX.ICMP.Message;
+            use RFLX.IPv4.Packet;
+            use RFLX.RFLX_Builtin_Types;
+         begin
+            pragma Assert (ICMP_Context.Last = 84 * 8 - 32 * 5 + Field_First (IP_Context, F_Payload) - 1);
+            RFLX.ICMP.Message.Set_Tag (ICMP_Context, RFLX.ICMP.Echo_Request);
+            RFLX.ICMP.Message.Set_Code_Zero (ICMP_Context, 0);
+            RFLX.ICMP.Message.Set_Checksum (ICMP_Context,
+                                            Checksum.Echo_Request_Reply_Checksum
+                                               (RFLX.ICMP.Echo_Request,
+                                                0, 0, Sequence, Data));
+            RFLX.ICMP.Message.Set_Identifier (ICMP_Context, 0);
+            RFLX.ICMP.Message.Set_Sequence_Number (ICMP_Context, Sequence);
+            pragma Assert (Predecessor (ICMP_Context, F_Data) = F_Sequence_Number);
+            pragma Assert (Field_Last (ICMP_Context, F_Sequence_Number) -
+                              Field_First (IP_Context, F_Payload) + 1
+                           = 8 + 8 + 16 + 16 + 16);
+            pragma Assert (Field_First (ICMP_Context, F_Data) -
+                              RFLX.IPv4.Packet.Field_First (IP_Context, F_Payload)
+                           = 8 + 8 + 16 + 16 + 16);
+            RFLX.ICMP.Message.Set_Data (ICMP_Context, Data);
+         end;
```

Comment:
I think it is a bit awkward to have such assertions in an example. This gives the impression that the generated code is very hard to use, at least if you want to be able to prove it. I think it is fine to make the generated code complex, but we should try to make the user code as simple as possible. This will also help us for the session code generation, where we also use this API. Do you see any alternatives?

Reply:
I added those assertions before simplifying some preconditions, so maybe we can remove those now. To be checked.

Reply:
Just for info, the code is proved without any assertions, but the precondition of Set_Data needs almost 3 minutes on my machine, and even more in the CI. The 4 assertions bring it down to less than 30 seconds for the longest proof.
I was able to remove 3 of the 4 assertions and the longest time is now ~60s (on my machine).

Reply:
We decided in our meeting to aim for the removal of all remaining assertions. The declare blocks could be removed by moving the use clauses into the declarative part of the subprogram.

Reply:
Current status is that I still need one assertion. I simplified it and removed the declare blocks.

Reply:
Just one assertion left. Can we leave this in and create a separate ticket to remove the assertion? I agree with the general idea that we want to minimize the amount of assertions in such examples.

Reply:
> Just one assertion left. Can we leave this in and create a separate ticket to remove the assertion?

Yes, let's do that.

Reply:
OK, I'm resolving this conversation.